src/debug.cc

```diff
@@ -421,24 +421,25 @@ syslog_facility_names[] = {
     }
 };
 
-#endif
+#endif /* LOG_LOCAL4 */
+#endif /* HAVE_SYSLOG */
 
 void
 _db_set_syslog(const char *facility)
 {
+#if !HAVE_SYSLOG
+    fatal("Logging to syslog not available on this platform");
+#endif
     Debug::log_syslog = true;
 
-#ifdef LOG_LOCAL4
-#ifdef LOG_DAEMON
-
-    syslog_facility = LOG_DAEMON;
-#else
-
+#if defined(LOG_LOCAL4)
     syslog_facility = LOG_LOCAL4;
-#endif /* LOG_DAEMON */
+#elif defined(LOG_DAEMON)
+    syslog_facility = LOG_DAEMON;
+#endif
 
     if (facility) {
-
+#if HAVE_SYSLOG
         struct syslog_facility_name *n;
 
         for (n = syslog_facility_names; n->name; ++n) {
```

Comment:
The code is overly generic. I think it would now be safe to assume that if we have syslog, then all standard facilities are defined. Instead, there's a long list of #ifdefs that are quite painful to read.
I'll mention that we stop having special reliance on these two in the PR description.